conans/client/build/cppstd_flags.py

```diff
@@ -22,7 +22,10 @@ def cppstd_from_settings(settings):
     return compiler_cppstd or cppstd
 
 
-def cppstd_flag(compiler, compiler_version, cppstd):
+def cppstd_flag(settings):
```

Comment:
We cannot remove (yet) the old implementation or the `boost` recipe will be broken.

Reply:
IMO as it's not part of public API, so it's subject to change. boost recipe is already "broken" as it uses undocumented private API. we will fix the recipe right away as soon as new tool released.

Reply:
No, let's add some time buffer. We can release this `tool`, wait a couple of weeks and then modify the `boost` recipe.  Otherwise `boost` recipe in user's cache will be broken if they update the Conan client.

Reply:
okay, I've kept old implementation for a while.